.github/workflows/update-os-snapshots.yml

```diff
@@ -0,0 +1,250 @@
+name: Refresh E2E Snapshots
+on:
+  workflow_dispatch:
+
+jobs:
+  e2e-setup-login:
+    if: ${{ github.ref != 'refs/heads/main' }}
+    name: End-to-end login
+    runs-on: 'ubuntu-latest'
+    outputs:
+      cliConfigJson: ${{ steps.setup.outputs.cliConfigJson}}
+    env:
+      # Platform environment to log into for the e2e tests.
+      PLATFORM_ENV: 'stg'
+      # Username used to log into the organization whose ID is stored in the ORG_ID variable
+      PLATFORM_USER_NAME: ${{ secrets.PLATFORM_USER_NAME }}
+      # Password used to log into the organization whose ID is stored in the ORG_ID variable
+      PLATFORM_USER_PASSWORD: ${{ secrets.PLATFORM_USER_PASSWORD }}
+      # API key to delete all the API keys created by the e2e tests
+      PLATFORM_API_KEY: ${{ secrets.PLATFORM_API_KEY }}
+      # ID of the organization to log into for the e2e tests
+      ORG_ID: ${{ secrets.ORG_ID }}
+      # ID of the test run to identify resources to teardown.
+      TEST_RUN_ID: 'id${{ matrix.os }}-${{ github.sha }}-${{ github.run_attempt }}g'
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # tag=v3
+      - uses: actions/setup-node@2fddd8803e2f5c9604345a0b591c3020ee971a93 # tag=v3
+        with:
+          node-version: ${{matrix.node}}
+          cache: 'npm'
+      - name: Install locked NPM version.
+        # TODO: CDX-1010 | Remove in CDX-1010.
+        run: npm i -g npm@8.10.0
+      - run: |
+          npm ci
+          Xvfb :1 -screen 0 1024x768x16 & sleep 1
```

Comment:
II think a small comment that explains why these are needed for chrome to work properly (I assume?) would be nice ;)